### PR TITLE
Add method to remove callback from secure sockets implimentation

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
In reference to https://github.com/aws/amazon-freertos/issues/1646, this PR fixes the callback removal issue which could cause an orphan task to be created.
Cleaning a callback requires the spawned task to be terminated before everything can be cleaned up. This requires two way communication/synchronization. Thus, event groups were used to allow both tasks to communicate with each other of the actions required.
Also, the `lwip_select` function was made to block for 10 seconds instead of infinite time - this would introduce overheads but was necessary. The time interval of 10 seconds can be changed by modifying the provided macro. 

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c [hunk 1]
```diff
@@ -319,6 +339,12 @@ static void prvRxSelectSet( ss_ctx_t * ctx,
     ctx->rx_callback = ( void ( * )( Socket_t ) )pvOptionValue;
 
     prvIncrementRefCount( ctx );
+
+    ctx->rx_EventGroup = NULL;
```
Comment (dan4thewin) [inline libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c:343]: Does the next line make the last unnecessary?

File: libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c [hunk 2]
```diff
@@ -336,7 +362,28 @@ static void prvRxSelectSet( ss_ctx_t * ctx,
...
+    /* Wait for the task to delete itself. */
+    if( xEventGroupWaitBits( ctx->rx_EventGroup, SOCKETS_COMPLETE_DELETION, pdTRUE, pdTRUE, portMAX_DELAY ) == SOCKETS_COMPLETE_DELETION )
+    {
+        /* The task 'vTaskRxSelect' has started deleting itself. */
+    }
+    else
+    {
+        /* Control should never reach here as this task should wait on the event group bits indefinately. */
+        configASSERT( pdTRUE );
+    }
```
Comment (dan4thewin) [inline libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c:377]: I think this is an assumption.  I think indefinite blocking requires INCLUDE_vTaskSuspend.  If a customer configures that to be 0, then the function could return due to timeout.  Perhaps wrap this in a while loop?

File: libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c (hunk 1 above)
Comment (AniruddhaKanhere) [reply libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c]: Yes, I missed that. Thanks, fixed it.

File: libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c (hunk 2 above)
Comment (AniruddhaKanhere) [reply libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c]: Ah yes! I overlooked that. I have changed that to a while. Thanks for catching that Dan.

**Review by AniruddhaKanhere — COMMENTED** [review]
Thanks for the review Dan. I have applied your suggestions.

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**